tools/runfiles/runfiles.rs

```diff
@@ -44,36 +44,49 @@ const MANIFEST_FILE_ENV_VAR: &str = "RUNFILES_MANIFEST_FILE";
 const MANIFEST_ONLY_ENV_VAR: &str = "RUNFILES_MANIFEST_ONLY";
 const TEST_SRCDIR_ENV_VAR: &str = "TEST_SRCDIR";
 
+#[macro_export]
+macro_rules! Rlocation {
```

Comment:
I'm not going to say we should do it one way or the other, but we should definitely at least consider the folllowing API as an alternative:

```
let  r = runfiles::create!() // Expands to runfiles::create(env!("REPOSITORY_NAME"))
r.rlocation(path)
```

Pros:
* More clean API (less macros)
* More performant (only need to calculate the repo mapping once)
* Simpler to migrate to (only migrate the runfiles constructor, not the rlocation calls)

Cons:
* If you're passing the runfiles constructor across repo boundaries, it might do the wrong repo mapping. I can't see this ever happening though.

Reply:
Yep I asked @fmeum the same and he mentioned passing across repo boundaries is a nice to have. I'm fine either but let's figure it out in this thread before I refactor more :)

Re: more performant - I think it's the same? The mapping is only calculated at startup? Or if you're referring to the map lookups + string ops - you're about to do File IO, I don't think it matters :)

The migration path is also a good point. I think as-written the code doesn't _require_ any client changes, but they can move to the new API to become more correct. I think with your proposal everyone using runfiles needs a minor code change?

Reply:
Ah, good point about the migration path, I hadn't notice that we left the old method, so I thought this whole PR was backwards incompatible. If that's the case, I think we definitely just do it the way you've already done it.